### PR TITLE
Added mocked test of status codes

### PR DESCRIPTION
**Pull request by jacobcook1995**
Added a basic test to check that errors are correctly raised when certain status codes are returned. It's fairly bare bones at the moment, but I thought it was useful functionality to add as I might want to include more complex http status code handling in future, and the test can be extended to cover future cases. It didn't seem worth it to me to test `taxonomy_esearch` and `taxonomy_efetch` separately as they are very similar, so the test tests both of them at once.

### CODE REVIEW

File: test/test_ncbi_taxa.py [hunk 1]
```diff
@@ -24,6 +25,30 @@ def test_sdv_no_remote_set_correctly():
...
+    # Configure the mock to return a response with a specific status code.
+    mock_get.return_value.status_code = status_code
+
+    # Construct remote validator
+    v = taxa.RemoteNCBIValidator(resources_with_remote_ncbi)
+
+    # Call the service, which will send a request to the server.
+    with pytest.raises(expected_exception[1], match=expected_exception[0]):
+        _ = v._taxonomy_efetch(0)
+        _ = v._taxonomy_esearch("NAME")
```
Comment (dalonsoa) [inline test/test_ncbi_taxa.py:49]: If you aren't using the returned values, just don't assign them to anything.
```suggestion
        v._taxonomy_efetch(0)
        v._taxonomy_esearch("NAME")
```

**Review by dalonsoa — APPROVED** [review]
It looks good to me! This is the spirit of mocking1

**Comment by jacobcook1995** [comment]
Made the change you suggested, + switched to using valid input so that if mocking somehow broke down the test wouldn't still pass. I'll leave the pull request open for David to look at when he gets back from holiday

**Comment by davidorme** [comment]
It seems a bit strange to me to be using `unittest.mock` here, rather than using the `pytest-mock` extension, which  provides the `mocker` fixture for use in unit tests.  Digging into it, it seems like `pytest-mock` is actually just exposing a fixture interface to the `mock` library, which became part of the standard library as `unittest.mock`, so they are effectively just different APIs to the exact same functionality.

It seems like the `unittest` approach _works_ , and the decorator helps emphasize _what_ is being mocked in a test, but using a single framework seems cleaner? @dalonsoa - thoughts?

**Comment by dalonsoa** [comment]
I don't have strong feelings either way, to be honest. I've used `unittest` in many projects without issues, and in others I've used `pytest-mock` also fine. I don't think it matters as long as you are consistent, using ONLY one or the other within a project.

**Review by dalonsoa — APPROVED** [review]
Looks good to me!

**Pull request merged**